### PR TITLE
extract building of UnsignedErgoLikeTransaction to use in playgrounds

### PR DESCRIPTION
**Pull request by greenhat**

### CODE REVIEW

File: lib-impl/src/main/java/org/ergoplatform/appkit/impl/TransactionOperations.scala [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    )
+
+    val tx = new UnsignedErgoLikeTransaction(
+      inputs.map(b => new UnsignedInput(b.id)),
+      dataInputs,
+      finalOutputCandidates.toIndexedSeq
+    )
+    tx
+  }
+}
```
Comment (aslesarenko) [inline lib-impl/src/main/java/org/ergoplatform/appkit/impl/TransactionOperations.scala]: @greenhat this code looks like belonging to sigmastate library, i.e. it mostly uses Sigma classes.
But I wonder how many of such methods we will need on Sigma side? There needs to be some systematic approach for creating them. And also how to make this methods stable, i.e. they should cover all possible transactions, withing our underlying transaction model.

File: lib-impl/src/main/java/org/ergoplatform/appkit/impl/UnsignedTransactionBuilderImpl.java [hunk 2]
```diff
@@ -85,53 +86,16 @@ public UnsignedTransactionBuilder sendChangeTo(ErgoAddress changeAddress) {
...
-        UnsignedErgoLikeTransaction tx =
-                new UnsignedErgoLikeTransaction(inputs, dataInputs, outputCandidates);
-        List<ErgoBox> boxesToSpend =
-                _inputBoxes.stream().map(b -> b.getErgoBox()).collect(Collectors.toList());
+        IndexedSeq<ErgoBox> inputBoxes = JavaHelpers.toIndexedSeq(boxesToSpend);
+        UnsignedErgoLikeTransaction tx = TransactionOperations.buildUnsignedErgoTx(inputBoxes, dataInputs,
+                outputCandidates, _feeAmount, _changeAddress, _ctx.getHeight());
         ErgoLikeStateContext stateContext = createErgoLikeStateContext();
 
         return new UnsignedTransactionImpl(tx, boxesToSpend, new ArrayList<>(), stateContext);
```
Comment (aslesarenko) [inline lib-impl/src/main/java/org/ergoplatform/appkit/impl/UnsignedTransactionBuilderImpl.java]: @greenhat As long as `TransactionOperations.buildUnsignedErgoTx` is moved to Sigma this code is ok.

File: lib-impl/src/main/java/org/ergoplatform/appkit/impl/TransactionOperations.scala (hunk 1 above)
Comment (greenhat) [reply lib-impl/src/main/java/org/ergoplatform/appkit/impl/TransactionOperations.scala]: @aslesarenko I'd rather refrain from adding stuff to sigma, that sigma itself does not need. It's already huge.
We can put them into separate lib, or add to the wallet(we already have various box operations there).

Comment (aslesarenko) [reply lib-impl/src/main/java/org/ergoplatform/appkit/impl/TransactionOperations.scala]: Ah, yes wallet is better. It looks also related to the issue https://github.com/ergoplatform/ergo/issues/1029

**Comment by greenhat** [comment]
Created #44

**Comment by greenhat** [comment]
Done in #46

**Pull request closed without merging**